hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/util/ReflectionUtilsTest.java

```diff
@@ -50,9 +63,49 @@ public void readStaticFieldOrNull_readFromPublicField() {
         assertEquals("staticPublicFieldContent", field);
     }
 
+    @Test
+    public void shouldDiscoverAllNestedClasses() throws ClassNotFoundException {
+        // When
+        Collection<Class<?>> classes = ReflectionUtils.nestedClassesOf(OuterClass.class);
+
+        // Then
+        assertThat(classes, hasSize(3));
+        assertThat(classes, containsInAnyOrder(
+                OuterClass.class,
+                OuterClass.NestedClass.class,
+                Class.forName("com.hazelcast.jet.impl.util.ReflectionUtilsTest$OuterClass$1")
+        ));
+    }
+
+    @Test
+    public void shouldDiscoverAllClassesAndResourcesInAPackage() {
```

Comment:
This should assert that the nested class is also there.

Reply:
Then we would actually test whether JVM creates nested classes in the same package as the root class which is not really the goal here.

Reply:
It definitely is in the same package, and if it's not included, then `addPackages` is broken, going back to the place we were before this PR.

Reply:
What I mean is - there are many more classes in this package, the nested class is just one of them, in the context of this test nested class is no different from the _regular_ class, I guess to properly test it I should create a separate package with known classes in it and ensure all of them are listed, I was thinking about it but then didn't like the idea of creating yet another package... WDYT?

Reply:
My thinking was explicitly look for the nested classes that you already look for in `addClass`.